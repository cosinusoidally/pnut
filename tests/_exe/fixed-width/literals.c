#include <stdio.h>

void putstr(char* s) {
  while (*s) {
    putchar(*s);
    s++;
  }
}

void main() {
  int a = 0x12345678;

  // Tests that casts properly truncate the value
  if (a == (char) a) {
    putstr("!!! a == (char) a\n");
  } else {
    putstr("a != (char) a\n");
  }

  // Again
  if (a == (short) a) {
    putstr("!!! a == (short) a\n");
  } else {
    putstr("a != (short) a\n");
  }

  // Again, but this time the cast is to int so it should be equal
  if (a == (int) a) {
    putstr("a == (int) a\n");
  } else {
    putstr("!!! a != (int) a\n");
  }

  // Again, but with a cast to a wider type
  if (a == (long) a) {
    putstr("a == (long) a\n");
  } else {
    putstr("!!! a != (long) a\n");
  }
}
