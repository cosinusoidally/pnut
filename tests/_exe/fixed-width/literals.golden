a != (char) a
a != (short) a
a == (int) a
a == (long) a
